scripts/polygon-eth/token_transfer.py

```diff
@@ -0,0 +1,316 @@
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+from web3 import Web3
+from eth_account import Account
+from web3.middleware import geth_poa_middleware
+from ledgereth import get_accounts, create_transaction, sign_transaction
+from ledgereth.web3 import LedgerSignerMiddleware
+import requests
+import json
+import os
+import binascii
+import sys, getopt
+import time
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+ALCHEMY_API_KEY_ETH = os.getenv("ALCHEMY_API_KEY_ETH")
+ALCHEMY_API_KEY_POLYGON = os.getenv("ALCHEMY_API_KEY_POLYGON")
+ETHERSCAN_API_KEY = os.getenv("ETHERSCAN_API_KEY")
+POLYGONSCAN_API_KEY = os.getenv("POLYGONSCAN_API_KEY")
+PRIVATE_KEY = os.getenv("PRIVATE_KEY")
+
+# Get the config JSON
+with open("config.json") as f:
+    config = json.load(f)
+# Search for unfinished transfers on L1
+from_block_l1 = config['from_block_l1']
+from_block_l2 = config['from_block_l2']
+# Proof generator URL and message hash topic
+proof_generator_url = config['proof_generator_url']
+msg_hash_topic = "0x8c5261668696ce22758910d05bab8f186d6eb247ceac2af2e82c7dc17669b036"
+
+w3_l1 = Web3(Web3.HTTPProvider(f"{config['alchemy_url_eth']}{ALCHEMY_API_KEY_ETH}"))
+w3_l2 = Web3(Web3.HTTPProvider(f"{config['alchemy_url_polygon']}{ALCHEMY_API_KEY_POLYGON}"))
+w3_l2.middleware_onion.inject(geth_poa_middleware, layer=0)
+ledger = config['ledger']
+account_address = ""
+
+# Get contract ABI
+def get_abi(contract_address, url_api, scan_key):
+    # Replace 'YourEtherscanApiKey' with your actual Etherscan API key
+    url = f"{url_api}/api?module=contract&action=getabi&address={contract_address}&apikey={scan_key}"
+    response = requests.get(url)
+    data = response.json()
+    if data['status'] == '1':
+        return json.loads(data['result'])
+    else:
+        raise ValueError(data['result'])
+
+# Send the tx
+def send_tx(message, tx, w3):
+    if ledger:
+        txn = w3.eth.send_transaction(
+            {
+                "from": account_address,
+                "to": tx['to'],
+                "value": tx['value'],
+                "gas": tx['gas'],
+                "maxPriorityFeePerGas": tx['maxPriorityFeePerGas'],
+                "maxFeePerGas": tx['maxFeePerGas'],
+                "data": tx['data'],
+                "chain_id": tx['chainId']
+            }
+        )
+    else:
```

Comment:
This is an unreachable branch, as you are accessible the config using the key of the dictionary, it can never ever return None, it will throw instead

Reply:
Disagree, see above